### PR TITLE
`no-array-for-each`: Stop mention "Array"

### PR DESCRIPTION
**Pull request by fisker**
Stop calling the object "array" in code and docs. #1778

### CODE REVIEW

File: docs/rules/no-array-for-each.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Prefer `for…of` over `Array#forEach(…)`
+# Prefer `for…of` over `forEach` method
```
Comment (sindresorhus) [inline docs/rules/no-array-for-each.md:1]:
```suggestion
# Prefer `for…of` over the `forEach` method
```

File: docs/rules/no-array-for-each.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 🔧💡 *This rule is [auto-fixable](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) and provides [suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions).*
 <!-- /RULE_NOTICE -->
 
-Benefits of [`for…of` statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) over [`Array#forEach(…)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) can include:
+Benefits of [`for…of` statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) over `forEach` method can include:
```
Comment (sindresorhus) [inline docs/rules/no-array-for-each.md:10]:
```suggestion
Benefits of [`for…of` statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) over the `forEach` method can include:
```

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by fisker** [comment]
Merging this, so I can work on improving this rule.

**Pull request merged**